bids/layout/layout.py

```diff
@@ -206,19 +206,26 @@ def __init__(self, root, validate=True, absolute_paths=True,
         self.sources = sources
         self.regex_search = regex_search
         self.config_filename = config_filename
+        # Store init arguments for saving
+        self._sanitize_init_args(
```

Comment:
Probably better to return the `init_args` and set them here, so it's clear from the `__init__` what the instance attributes are.

Reply:
All of those get saved into a single instance attribute `init_args`. The reason for that is that I wanted to capture the actual passed arguments, not the instance attributes that may be further modified (e.g. `derivatives`). Especially for when the user calls `save` later on.

I can move that code to here to make it more transparent (and add a comment), but still think its best to save as a single dict with original unprocessed arguments.

Reply:
I just meant that the dict you construct in `_sanitize_init_args` should be returned from the method and set here rather than there. No objection to anything else.

Reply:
Done and done